Rocket.Chat/Controllers/Chat/ChatViewController.swift

```diff
@@ -207,6 +209,30 @@ final class ChatViewController: SLKTextViewController {
         ThemeManager.addObserver(navigationController?.navigationBar)
         textInputbar.applyTheme()
         chatTitleView?.state = SocketManager.sharedInstance.state
+
+        if self.subscription == nil {
+            title = ""
+            textView.isEditable = false
+            leftButton.isEnabled = false
+
+            chatTitleView?.removeFromSuperview()
+            backgroundImageViewEmptyState?.removeFromSuperview()
+
+            guard let theme = view.theme else { return }
+            let themeName = ThemeManager.themes.first { $0.theme == theme }?.title
+
+            let backgroundImageView = UIImageView(image: UIImage(named: "Empty State \(themeName ?? "light")"))
+            backgroundImageView.contentMode = .scaleAspectFill
+            backgroundImageView.clipsToBounds = true
+            self.view.insertSubview(backgroundImageView, belowSubview: textInputbar)
+
+            backgroundImageViewEmptyState = backgroundImageView
+            updateEmptyBackgroundImageFrames()
+        } else {
+            backgroundImageViewEmptyState?.removeFromSuperview()
+            textView.isEditable = true
+            leftButton.isEnabled = true
+        }
```

Comment:
I think you should put all this if-else in a separate method eg `showEmptyState(Bool)` so we don't bloat this setter 🙂